fili-core/src/test/groovy/com/yahoo/bard/webservice/data/QueryBuildingTestingResources.groovy

```diff
@@ -371,37 +285,40 @@ public class QueryBuildingTestingResources {
                 simpleTemplateWithGrainQuery,
                 WEEK
         )
-        setupPartialData()
+
+        return this
+        /* Example:
+        List<Column> columns = []
+        ConcretePhysicalTable table1 = ConcretePhysicalTable("name", DAY, columns, [:] )
+
+        Map<Column, List<Interval>> availableMap = columns.collectEntries { [(it): [interval2] ]}
+        table1.setAvailability(new ImmutableAvailability(availableMap))
+         */
     }
 
     def setupPartialData() {
         // In the event of partiality on all data, the coarsest table will be selected and the leftmost of the
         // coarsest tables should be selected
-        emptyFirst = new PhysicalTable("emptyFirst", MONTH.buildZonedTimeGrain(UTC), [:])
-        emptyLast = new PhysicalTable("emptyLast", MONTH.buildZonedTimeGrain(UTC), [:])
-        partialSecond = new PhysicalTable("partialSecond", MONTH.buildZonedTimeGrain(UTC), [:])
-        wholeThird = new PhysicalTable("wholeThird", MONTH.buildZonedTimeGrain(UTC), [:])
-
-        Interval emptyInterval = new Interval("2015/2015");
-        emptyFirst.addColumn(DimensionColumn.addNewDimensionColumn(emptyFirst, d1), [emptyInterval] as Set)
-        emptyLast.addColumn(DimensionColumn.addNewDimensionColumn(emptyLast, d1), [emptyInterval] as Set)
-        Interval oneYear = new Interval("2015/2016");
-        partialSecond.addColumn(DimensionColumn.addNewDimensionColumn(partialSecond, d1), [oneYear] as Set)
-        Interval fiveYears = new Interval("2011/2016")
-        wholeThird.addColumn(DimensionColumn.addNewDimensionColumn(wholeThird, d1), [fiveYears] as Set)
-
-        [m1, m2, m3].each {
-            emptyFirst.addColumn(MetricColumn.addNewMetricColumn(emptyFirst, it.name), [emptyInterval] as Set)
-            emptyLast.addColumn(MetricColumn.addNewMetricColumn(emptyLast, it.name), [emptyInterval] as Set)
-            partialSecond.addColumn(MetricColumn.addNewMetricColumn(partialSecond, it.name), [oneYear] as Set)
-            wholeThird.addColumn(MetricColumn.addNewMetricColumn(wholeThird, it.name), [fiveYears] as Set)
+        emptyFirst = new ConcretePhysicalTable("emptyFirst", MONTH.buildZonedTimeGrain(UTC), [d1, m1, m2, m3].collect{toColumn(it)}.toSet(), [:])
+        emptyLast = new ConcretePhysicalTable("emptyLast", MONTH.buildZonedTimeGrain(UTC), [d1, m1, m2, m3].collect{toColumn(it)}.toSet(), [:])
+        partialSecond = new ConcretePhysicalTable("partialSecond", MONTH.buildZonedTimeGrain(UTC), [d1, m1, m2, m3].collect{toColumn(it)}.toSet(), [:])
+        wholeThird = new ConcretePhysicalTable("wholeThird", MONTH.buildZonedTimeGrain(UTC), [d1, m1, m2, m3].collect{toColumn(it)}.toSet(), [:])
+
+        Map<Column, List<Interval>> availabilityMap1 = [:]
+        Map<Column, List<Interval>> availabilityMap2 = [:]
+        Map<Column, List<Interval>> availabilityMap3 = [:]
```

Comment:
Can we give these better names? "1", "2", "3" isn't very clear about what these will hold.